### ISSUE TITLE
Expose c-ares init parameters to the cluster api

### ISSUE THREAD

**Issue by davinci26** [envoyproxy/envoy#17795]
*Title*: *Expose c-ares init parametersto the cluster api*

*Description*:
>Describe the desired behavior, what scenario it enables and how it
would be used.

Similar to `use_tcp_for_dns_lookups` [docs](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/resolver.proto#envoy-v3-api-msg-config-core-v3-dnsresolutionconfig) I would also like the timeouts (see [ares_init_options](https://c-ares.haxx.se/ares_init_options.html)) property to be exposed via the cluster api.

### Why this change is important

tl;dr: when a Windows container startup the first channel is dirty and this incurs a 2:30 mins timeout.

Consider the following setup:

Kubernetes cluster with a Windows and Linux node

On linux node: control.plane.svc.cluster.local
Windows:

1. Envoy container is trying to start and attempts to resolve address control.plane.svc.cluster.*local*
2. Envoy attempts to resolve the address
3. c-ares is loads the known dns suffix from registry keys into the channel when
4. c-ares attempts to resolve the address with timeout t and retries r.
5. Windows is setting up the container registry with the known suffix.
6. c-ares never updates the channel and the dns resolution times out after two minutes

**Comment by davinci26** [comment]
@envoyproxy/api-shepherds since this involves an API change

@envoyproxy/windows-dev

**Comment by htuch** [comment]
Can we do this after https://github.com/envoyproxy/envoy/pull/17479 lands and restrict to just the c-ares extension?

**Comment by davinci26** [comment]
yes, for sure, makes sense. I don't mind waiting for this PR to get merged as long as it is within a reasonable time.

Is the PR on track to get merged or are you iterating on the design?

**Comment by davinci26** [comment]
This issue that motivated the change appears only when the container runtime is docker rather than containerd.

According to sig-windows Kubernetes is moving forward with containerd as the container runtime in the future so I do not have motivation for this change.

If anyone needs this feature then feel free to open the issue and I can chime in
